public/app/features/dashboard/components/ShareModal/ShareLink.test.tsx

```diff
@@ -65,149 +67,104 @@ jest.mock('@grafana/runtime', () => {
   };
 });
 
-interface ScenarioContext {
-  wrapper?: ShallowWrapper<Props, State, ShareLink>;
-  mount: (propOverrides?: Partial<Props>) => void;
-  setup: (fn: () => void) => void;
-}
-
-function shareLinkScenario(description: string, scenarioFn: (ctx: ScenarioContext) => void) {
-  describe(description, () => {
-    let setupFn: () => void;
-
-    const ctx: any = {
-      setup: (fn: any) => {
-        setupFn = fn;
-      },
-      mount: (propOverrides?: any) => {
-        const props: any = {
-          panel: undefined,
-          dashboard: { time: getDefaultTimeRange() },
-        };
-
-        Object.assign(props, propOverrides);
-        ctx.wrapper = shallow(<ShareLink {...props} />);
-      },
-    };
-
-    beforeEach(() => {
-      setUTCTimeZone();
-      setupFn();
-    });
-
-    scenarioFn(ctx);
-  });
-}
-
 describe('ShareModal', () => {
   let templateSrv = initTemplateSrv('key', []);
+  let props: Props;
 
   beforeAll(() => {
     setEchoSrv(new Echo());
     variableAdapters.register(createQueryVariableAdapter());
     setTemplateSrv(templateSrv);
   });
 
-  shareLinkScenario('shareUrl with current time range and panel', (ctx) => {
-    ctx.setup(() => {
-      mockLocationHref('http://server/#!/test');
-      config.bootData = {
-        user: {
-          orgId: 1,
-        },
-      } as any;
-      ctx.mount({
-        panel: new PanelModel({ id: 22, options: {}, fieldConfig: { defaults: {}, overrides: [] } }),
-      });
-    });
+  beforeEach(() => {
+    setUTCTimeZone();
+    mockLocationHref('http://server/#!/test');
+    config.rendererAvailable = true;
+    config.bootData.user.orgId = 1;
+    props = {
+      panel: new PanelModel({ id: 22, options: {}, fieldConfig: { defaults: {}, overrides: [] } }),
+      dashboard: new DashboardModel({ time: getDefaultTimeRange(), id: 1 }),
+    };
+  });
 
+  describe('with current time range and panel', () => {
     it('should generate share url absolute time', async () => {
-      await ctx.wrapper?.instance().buildUrl();
-      const state = ctx.wrapper?.state();
-      expect(state?.shareUrl).toBe('http://server/#!/test?from=1000&to=2000&orgId=1&viewPanel=22');
+      render(<ShareLink {...props} />);
+      expect(await screen.findByLabelText('Link URL')).toHaveValue(
```

Comment:
Personally I'd use `await screen.findByRole('textbox', { name: 'Link URL' })`, because [it's more robust](https://testing-library.com/docs/queries/bylabeltext/#name), but I guess this works as well.